src/stages/stage5/timeline.c

```diff
@@ -37,537 +26,845 @@ static void stage5_dialog_post_midboss(void) {
 	INVOKE_TASK_INDIRECT(Stage5PostMidBossDialog, pm->dialog->Stage5PostMidBoss);
 }
 
-static int stage5_greeter(Enemy *e, int t) {
-	TIMER(&t)
-	AT(EVENT_KILLED) {
-		spawn_items(e->pos, ITEM_POINTS, 2, ITEM_POWER, 2);
-		return 1;
-	}
-
-	e->moving = true;
-	e->dir = creal(e->args[0]) < 0;
-
-	FROM_TO(0, 50, 1) {
-		GO_TO(e, e->pos0 + e->args[0] * 50, 0.05);
-	}
-
-	if(t > 200)
-		e->pos += e->args[0];
-
-	FROM_TO(80, 180, 20) {
-		for(int i = -(int)global.diff; i <= (int)global.diff; i++) {
-			PROJECTILE(
-				.proto = pp_bullet,
-				.pos = e->pos,
-				.color = RGB(0.0, 0.0, 1.0),
-				.rule = asymptotic,
-				.args = {
-					(3.5+(global.diff == D_Lunatic))*cexp(I*carg(global.plr.pos-e->pos) + 0.06*I*i),
-					5
-				}
-			);
-		}
-
-		play_sound("shot1");
-	}
-
-	return 1;
-}
-
-static int stage5_lightburst(Enemy *e, int t) {
-	TIMER(&t);
-	AT(EVENT_KILLED) {
-		spawn_items(e->pos, ITEM_POINTS, 4, ITEM_POWER, 2);
-		return 1;
-	}
-
-	FROM_TO(0, 70, 1) {
-		GO_TO(e, e->pos0 + e->args[0] * 70, 0.05);
-	}
-
-	if(t > 200)
-		e->pos += e->args[0];
+static void midboss_dummy(Boss *b, int t) { }
 
-	FROM_TO_SND("shot1_loop", 20, 300, 5) {
-		int c = 5+global.diff;
-		for(int i = 0; i < c; i++) {
-			cmplx n = cexp(I*carg(global.plr.pos) + 2.0*I*M_PI/c*i);
+TASK(magnetto_swirl_shoot, {
+	BoxedEnemy e;
+	cmplx move_to;
+}) {
+	Enemy *e = TASK_BIND(ARGS.e);
+
+	int interval = difficulty_value(12, 9, 6, 3);
+	int bullets = difficulty_value(1, 2, 2, 2);
+	int bullet_dir = difficulty_value(0, 0, 1, 0);
```

Comment:
why is a direction valued 0 and 1?

Reply:
There's a few places in the original code where it specifically checks against `D_Hard` and `D_Lunatic` being defined. Example:

```
			PROJECTILE(
				.proto = pp_ball,
				.pos = e->pos,
				.color = RGBA(0.1 + 0.5 * pow((t - 140) / 140.0, 2), 0.0, 0.8, 0.0),
				.rule = accelerated,
				.args = {
					(-2 + (global.diff == D_Hard)) * dir, // <--- this line
					0.02 * dir * (!i || global.diff != D_Lunatic), // <--- also this line
				},
```

It seems a bit silly to me, but I was trying to replicate original behaviour as much as possible.